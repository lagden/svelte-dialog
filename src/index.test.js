// @vitest-environment jsdom
import {describe, it, expect, beforeAll, vi} from 'vitest'
import {Dialog, acts} from '$lib/index.js'

beforeAll(() => {
	HTMLDialogElement.prototype.show = vi.fn()
	HTMLDialogElement.prototype.showModal = vi.fn()
	HTMLDialogElement.prototype.close = vi.fn()
})

describe('dom', () => {
	it('jsdom', () => {
		expect(typeof globalThis).not.toBe('undefined')
	})

	it('dialog', () => {
		new Dialog({target: globalThis.document.body})
		acts.open()
		expect(globalThis.document.body).toMatchSnapshot()
	})
})
