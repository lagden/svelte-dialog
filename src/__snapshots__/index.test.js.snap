// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dom > dialog 1`] = `
<body>
  <dialog
    class="tadashi-svelte-dialog s-PNpndoCA_M6Y"
  >
    <h3>
      Missing component!
    </h3>
     
    <button>
      Close
    </button>
    
  </dialog>
  <!--&lt;Base&gt;-->
  <!--&lt;Dialog&gt;-->
</body>
`;
